internal/osbuild2/yum_config_stage.go

```diff
@@ -0,0 +1,62 @@
+package osbuild2
+
+import (
+	"fmt"
+)
+
+type YumConfigConfig struct {
+	HttpCaching *string `json:"http_caching,omitempty"`
+}
+
+type YumConfigPlugins struct {
+	Langpacks *YumConfigPluginsLangpacks `json:"langpacks,omitempty"`
+}
+
+type YumConfigPluginsLangpacks struct {
+	Locales []string `json:"locales"`
+}
+
+type YumConfigStageOptions struct {
+	Config  *YumConfigConfig  `json:"config,omitempty"`
+	Plugins *YumConfigPlugins `json:"plugins,omitempty"`
+}
+
+func (YumConfigStageOptions) isStageOptions() {}
+
+func (o YumConfigStageOptions) validate() error {
+	var err error = nil
+	// Allow values from the osbuild schema
+	if o.Config != nil && o.Config.HttpCaching != nil {
+		valid := false
+		allowed_http_caching_values := []string{"all", "packages", "lazy:packages", "none"}
+		for _, v := range allowed_http_caching_values {
+			if v == *o.Config.HttpCaching {
+				valid = true
+			}
+		}
+		if !valid {
+			err = fmt.Errorf("yum config parameter http_caching does not allow %s as a value", *o.Config.HttpCaching)
+		}
+	}
+
+	if o.Plugins != nil && o.Plugins.Langpacks != nil && len(o.Plugins.Langpacks.Locales) < 1 {
+		append := ""
+		if err != nil {
+			append = fmt.Sprintf(", %s", err.Error())
+		}
+		err = fmt.Errorf("locales must contain at least one element, %s", append)
+	}
+
+	return err
```

Comment:
And here you could return `nil`. I find it easier to understand what value is returned, than when a variable is used. I would argue that the important part here is that the validation failed, than to capture all the issues, especially when the code panics when it does. But I understand that capturing all errors adds some value. I don't have a strong opinion about this, but think that we should be consistent across the codebase.

Reply:
Hmm, I guess I'm fine either way, but this seemed like a better solution to me because you can see right away all the issues and you can possibly save yourself from doing 2 updates instead of a single one.

Reply:
Please have look in the new PR. All requested changes are there.